pkg/volume/downwardapi/downwardapi.go

```diff
@@ -262,6 +261,24 @@ func CollectData(items []v1.DownwardAPIVolumeFile, pod *v1.Pod, host volume.Volu
 	return data, utilerrors.NewAggregate(errlist)
 }
 
+// TODO: unify with Kubelet.podFieldSelectorRuntimeValue
```

Comment:
rather than duplicating podFieldSelectorRuntimeValue, this seems like the right opportunity to move that method (it doesn't actually depend on anything in `*Kubelet`) and make use of it consistently

Reply:
But I tried #64168 pod , and it can get the right status.podIP. I am not sure the difference from kubelet and from pod

Reply:
volume setup and container start are done in separate work queues. it is possible the information you are looking for is sometimes present when the volume setup is done, and sometimes not. need feedback from @kubernetes/sig-storage-pr-reviews and @kubernetes/sig-node-pr-reviews on whether we can depend on the pod object status at this point. see discussion at https://github.com/kubernetes/kubernetes/pull/42717#issuecomment-286635692 for the types of issues encountered

Reply:
also have  a concern about cycle import if using `Kubelet.podFieldSelectorRuntimeValue`

Reply:
Thanks @msau42 , I have tried, that's correct `I suspect what is happening here is the initial volume setup gets an empty IP, but sometime afterwards we periodically resync/remount the downward api volume, and it gets the updated IPs from the pod status.`

Reply:
ok

Reply:
>Did you check that hostIP doesn't have the same issue?

Yes,  podstatus generate before volume, pkg/kubelet/kubelet.go#L1433

>Is there somewhere in the code and/or docs we can document the reason why we can't support podIP in downward api volume?

No, I will add comment now.